packages/contractkit/src/utils/tx-signing.test.ts

```diff
@@ -1,56 +1,81 @@
 import debugFactory from 'debug'
+import * as util from 'util'
 import Web3 from 'web3'
+import { Provider } from 'web3/providers'
+import { generateAccountAddressFromPrivateKey } from '../providers/celo-private-keys-subprovider'
+import { CeloProvider } from '../providers/celo-provider'
 import { testWithGanache } from '../test-utils/ganache-test'
 import { recoverTransaction } from './signing-utils'
-import { CeloTx, getRawTransaction } from './tx-signing'
+import { CeloTx } from './tx-signing'
 import { addLocalAccount } from './web3-utils'
 
-// Run this test with --forceExit on completion.
-// jest src/utils/tx-signing.test.ts --forceExit
-
 const debug = debugFactory('kit:txtest:sign')
 
-// A random private key
-const PRIVATE_KEY = '1234567890abcdef1234567890abcdef1234567890abcdef1234567890abcdef'
-const accountAddress = generateAccountAddressFromPrivateKey(PRIVATE_KEY)
+// Random private keys
+const PRIVATE_KEY1 = '1234567890abcdef1234567890abcdef1234567890abcdef1234567890abcdef'
+const ACCOUNT_ADDRESS1 = generateAccountAddressFromPrivateKey(PRIVATE_KEY1)
+const PRIVATE_KEY2 = '1234567890abcdef1234567890abcdef1234567890abcdef1234567890fdeccc'
+const ACCOUNT_ADDRESS2 = generateAccountAddressFromPrivateKey(PRIVATE_KEY2)
+
+debug(`Private key 1: ${PRIVATE_KEY1}`)
+debug(`Account Address 1: ${ACCOUNT_ADDRESS1}`)
+debug(`Private key 2: ${PRIVATE_KEY2}`)
+debug(`Account Address 2: ${ACCOUNT_ADDRESS2}`)
 
-function generateAccountAddressFromPrivateKey(privateKey: string): string {
-  if (!privateKey.toLowerCase().startsWith('0x')) {
-    privateKey = '0x' + privateKey
+async function verifyLocalSigning(web3: Web3, from: string, to: string): Promise<void> {
+  const amountInWei: string = Web3.utils.toWei('1', 'ether')
+  const gasFees: string = Web3.utils.toWei('1', 'mwei')
+  debug('Signer Testing using Account: %s', from)
+  const celoTransaction: CeloTx = {
+    from,
+    to,
+    value: amountInWei,
+    gas: gasFees,
   }
-  // @ts-ignore-next-line
-  return new Web3.modules.Eth().accounts.privateKeyToAccount(privateKey).address
+  const signedTransaction = await web3.eth.signTransaction(celoTransaction)
+  debug('Singer Testing: Signed transaction %o', signedTransaction)
+  const rawTransaction: string = signedTransaction.raw
+  const recoveredSigner = recoverTransaction(rawTransaction)
+  debug('Transaction was signed by "%s", recovered signer is "%s"', from, recoveredSigner)
+  expect(recoveredSigner).toEqual(from)
 }
 
 testWithGanache('Transaction Utils', (web3: Web3) => {
-  // describe('Transaction Utils', () => {
-  describe.skip('Signer Testing', () => {
-    it('should be able to sign and get the signer back', async () => {
+  let originalProvider: Provider
+  beforeEach(() => {
+    originalProvider = web3.currentProvider
```

Comment:
I'm a bit lost here.. but I'm missing the part were we set the web3.currentProvider to CeloProvider. Where does that happen?

Reply:
see `addLocalAccount` in web3-utils.ts - https://github.com/celo-org/celo-monorepo/pull/798/files#diff-728b956ec7d19ed74a7cb0a6beb7be98

We change to CeloProvider when the first local account is added.